### PR TITLE
New room visibility format refactor

### PR DESCRIPTION
**Pull request by esh-g**
### Fixed Issues

$ https://github.com/Expensify/App/issues/32614
PROPOSAL: https://github.com/Expensify/App/issues/32614#issuecomment-1851610628

### Tests

1. Click on '+' FAB.
2. Click on 'Start chat'
3. Select the '#Room' tab
4. Enter any name
5. Hover over the visibility menu item
6. Make sure the bottom line of additional text is a part of the menu item. ('People in your workspace can find this room')
7. Click on the visibility.
8. Make sure the titles of all options are bold.
9. Make sure changing an option also changes the additional text of the menu item.

- [x] Verify that no errors appear in the JS console

### QA Steps

1. Click on '+' FAB.
2. Click on 'Start chat'
3. Select the '#Room' tab
4. Enter any name
5. Hover over the visibility menu item
6. Make sure the bottom line of additional text is a part of the menu item. ('People in your workspace can find this room')
7. Click on the visibility.
8. Make sure the titles of all options are bold.
9. Make sure changing an option also changes the additional text of the menu item.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/77237602/c4d3428f-56e0-49a3-b98d-b7d0c88dd2d4

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/77237602/a5db2c6c-634c-44d5-b64d-7611dcb3d6d0

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/77237602/599058d0-d12c-44d6-aa24-af91f80345bc

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/77237602/c0e725fd-0d51-443e-a3f6-c1ecf5199743

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/77237602/445b4ea5-68d9-44e9-a878-18d0a4194d9e

</details>

### CODE REVIEW

File: src/components/MenuItem.tsx
Comment (s77rt) [inline src/components/MenuItem.tsx:528]:
```suggestion
                                                {!!furtherDetailsIcon && (
```

The conditions here should always be boolean

**Review by s77rt — CHANGES_REQUESTED** [review]

**Comment by s77rt** [comment]
Font size does not match the mock up

Screenshot 2023-12-28 at 7 21 04 PM



Spacing does not match the mock up

Screenshot 2023-12-28 at 7 21 21 PM

**Comment by esh-g** [comment]
@s77rt As for the font-size in the menu item, it is the same style being used in the violations tab:
Screenshot 2023-12-29 at 11 05 00 AM

So, changing this would mean changes to this tab as well, it is possible to have the style props passed from the parent as separate but I think we should be consistent with how we handle all `furtherDetails` text in `MenuItem`. So, keeping all that in mind, should we change the `styles.furtherDetailsText`?

And does the spacing look fine now?

Screenshot 2023-12-29 at 11 03 53 AM

**Comment by s77rt** [comment]
@esh-g I think we make the changes to be global

**Comment by s77rt** [comment]
@shawnborton Need some help here:
1. Figma design for the selection list https://github.com/Expensify/App/issues/32614#issuecomment-1850797464

**Comment by s77rt** [comment]
@esh-g Yes, let's work on it then apply any necessary changes later

**Comment by esh-g** [comment]
> Yes, let's work on it then apply any necessary changes later
@s77rt So, right now we should wait for design approval before making any change or should I increase the text size now and then wait for design approval later?

**Comment by s77rt** [comment]
Let's increase the size first, I think the correct font size here is `variables.fontSizeLabel`

**Comment by esh-g** [comment]
Made the change, this is how it looks in violations:

Screenshot 2023-12-29 at 8 04 48 PM

**Comment by quinthar** [comment]
I'm sorry for my confusion here, can you catch me up?  So far as I can tell:

1. Is the question whether to change the global font size?  If so, I do *not* think we want to change any global fonts.  This is a very tiny, local change, let's follow the global styling, not change it.

2. Given that we're going to follow global stying, can you show what this looks like next to what the mockup asked for, side by side?  I suspect following global styling is just fine, but I'd like to see it.

Thanks!

**Comment by s77rt** [comment]
Mockup vs PR

289656893-9af14df1-e86a-4618-95ef-92598f047f18

289656893-9af14df1-e86a-4618-95ef-92598f047f18 copy



___

The "People in this workspace can find your room message" message shares the same stye with "Message hidden from channel, plus anonymous warning and message is reported for review" so you can see we had to increase the font size in order to match with the mockup. Do you think that increase in size is okay or we should keep that one untouched? 

293314302-1bf24570-dd62-4a02-97ec-1937ac515d43
293381691-d401bca7-9dfb-4ea6-8e10-e700657cdfcc

**Comment by quinthar** [comment]
Please leave the abuse screen untouched.  The PR looks fine to me; let's stick with the global styling.

**Comment by s77rt** [comment]
@quinthar I have opened a new PR for urgency (@esh-g does not seem online atm). https://github.com/Expensify/App/pull/33808 Can you merge it?

**Comment by quinthar** [comment]
Ah I will defer to others to merge.  Thanks for the hustle, though!

**Comment by esh-g** [comment]
@s77rt I am online now... should I just reverse the font change and merge this?

**Comment by quinthar** [comment]
Works for me!

**Comment by s77rt** [comment]
@esh-g No. I have created another PR https://github.com/Expensify/App/pull/33808 to keep the flag page untouched. Can you please close this one?

**Pull request closed without merging**
